Implement CopyFromRemote for io.Reader filesink

enmand opened the pull request:
This PR introduces CopyFromRemote (from #1) which returns an `io.Reader` with the file contents, the remote file permissions (as an int) and an error if it occurred.

Closes #1

enmand commented on client.go:
```diff
@@ -135,3 +198,79 @@ func (a *Client) Close() {
+	return errors.New("no scp cmd found")
+}
+
+func pipes(s *ssh.Session) (io.Reader, io.Reader, io.WriteCloser, error) {
```
`pipes` probably doesn't need to return `stderr`, but maybe it would be useful?

bramvdbogaerde commented:
This looks very interesting. 
I will try to review this by the end of the week.

enmand commented:
Hello! I've cleaned up the code to deal with permissions to use `os.FileMode` and properly use the octal representations, and added some tests for the Command Marshal and Unmarshal

bramvdbogaerde commented:
I am sorry for the late response.
Many thanks for your contributions.

I will take a look at it next week.

clarkezone commented:
any chance this will get merged soon?

bramvdbogaerde commented:
Yes, @clarkezone, I will try to make time to look at the changes this week.

hungarianguy commented:
@bramvdbogaerde @clarkezone is this something you plan to merge sometime soon?

bramvdbogaerde commented:
Indeed, it has been a while since this pull request was first created.

I think the idea is great, I do like the implementation but I have some small issues with some parts of the code, currently I don't have much time to carefully review the code and make some changes where necessary. I will try to do it by the end of the month.

clarkezone commented:
Any updates on this?

DrunkenPoney commented:
@bramvdbogaerde If you don't have time to maintain the project maybe you could add a maintainer to it. If you don't know anyone interested in becoming the new maintainer, you can create an issue and ask who is interested like [this one](https://github.com/gnunn1/tilix/issues/1700)

bramvdbogaerde commented:
I am sorry for my late response @DrunkenPoney, this project is certainly not dead, as can be demonstrated by my recent commits.

I plan to work on it more frequently now, and hope to have the suggested features added soon.

ceprateek commented:
when is this change going to be merged ? This is a handy functionality which is needed.

cpanato commented:
Hello Any plans to get this merged?

ChadTaljaardt commented:
Is this PR dead?

bramvdbogaerde commented:
I am sorry to keep you all waiting for so long.

I will try to review this pull request somewhere in the next few weeks and make 
some adjustments to keep up with the current code base if necessary.

tupyy commented:
Hello,

I'm playing around with this improvement because I need to write a lambda with is reading from a remote file. 
It is working fine but the final size of the copied file is one byte bigger than the original size.
Could you explained me why please?

Thanks a lot.

bramvdbogaerde commented:
At last, this was merged in PR #30, that PR has taken some inspiration from this PR, so many thanks for the changes here too.
